### ISSUE TITLE
Idea: Avoid combined video publishing and NFT minting

### ISSUE THREAD

**Issue by bedeho** [Joystream/atlas#4102]
# Background

This is a complex problem, and it is a consistent source of complexity and problems to deal with it for users when they need to edit something that was already minted for, or if the uplaod failed and they now need to attach an image which when cropped and resized always gives new hash, hence new metadata update transaction is needed, which fails. Here is one bug recently reported, which is costly to ameliorate properly

https://github.com/Joystream/atlas/issues/4101

I have also personally messed up a few videos this way, and it led me to stop minting NFTs for Jsgensis videos!

# Proposal

This needs a wiser solution end-to-end, but for right now, perhaps just hiding minting as part of publishing avoids the worst problems? Not sure how much that discourges minting. The real solution is this https://github.com/Joystream/atlas/issues/4103.

**Comment by attemka** [comment]
@KubaMikolajczyk 
Questions I have on this:

1. What should we do with the mint NFT toggle on the video creation?
2. How should we prevent assets changes during NFT mint?
3. How should we inform the user on mint NFT functionality (in case we'll hide\remove the toggle from video creation form)?

**Comment by dmtrjsg** [comment]
+ We need to prevent minting NFT if not all assets are in place (like thumbnail or description)
+ We need to remove the "Re-upload" thumbnail for the minted NFTs
+ we need to remove the NFT minting at the point of uploading videos

**Comment by KubaMikolajczyk** [comment]
I've explored possible solutions for this issue and in the short video below I outlined all of them together with benefits of my final recommendation ✅
🎥 Loom video: https://www.loom.com/share/95c05b850f1443ffb6ecda58b53baeb1?sid=f366e811-7f6a-4297-ab36-a4c5bdcadae5
📄 Figma explorations: https://www.figma.com/file/xm6vBdSnQDroc57rGfGtPM/Video-workspace?type=design&node-id=4817-127135&mode=design&t=3FJtqLr5Iq7D6rT2-4

**Comment by dmtrjsg** [comment]
Great stuff!

I think #3 is best for the manual uploads. For YPP syncs we certainly need the banner. More NFTs on sale are good for the users so we should not feel shy promoting this stronger. so modals from 3 with banner from 2 is my suggestion for this.

**Comment by KubaMikolajczyk** [comment]
@dmtrjsg @attemka @bedeho - NFT Minting & Video upload are separated successfully ✅ This introduced changes on 3 different pages but everything is explained in the video below and ready to be reviewed 👇

🎥Loom video: https://www.loom.com/share/c4de57758328491498ba83032244caa8?sid=820ad05b-c864-486d-880c-7e7141feb011

📄 My videos changelog: https://www.figma.com/file/LF7LOMhOA46bRcYHWlw1Bq/My-videos?type=design&node-id=784-26965&mode=design&t=JwSkHK3eNPGCm6fH-4

📄 Video workspace changelog: https://www.figma.com/file/xm6vBdSnQDroc57rGfGtPM/Video-workspace?type=design&node-id=1945-109809&mode=design&t=RvpLUU4Po4RYUFUn-4

📄 NFT Workspace changelog: https://www.figma.com/file/7LCeeLg9cCFSDAS4alvniX/NFT-Workspace?type=design&node-id=1-14416&mode=design&t=LfSGAE9xVoheDhcB-4

**Comment by drillprop** [comment]
You're wrong about thumbnails - they are **required**. In fact, one of the reasons why we have this issue was users can't reupload their thumbnails because something went wrong during thumbnail upload.  
The only optional fields are:
- description
- subtitles - rarely used
- published before - I think we can ignore this as well, also rarely used if any
So I'm not sure if the `mint video with missing assets` dialog still makes sense now. 

Also, there are no editable fields, after you mint an NFT. You can't 
- add subtitles
- disable/enable comments
- (playlists are not implemented)

because you'll receive an error.

**Comment by dmtrjsg** [comment]
Agreed with comments from @drillprop  and nice work on the bottom panels of new / edit states 👍

**Comment by traumschule** [comment]
Adding subtitles later on after feedback by the audience seems such an obvious and important feature even for NFT. Please make it possible somehow.

**Comment by dmtrjsg** [comment]
@traumschule NFTs are immutable on the runtime level, so not possible for the time being and comes with a long lead time..

**Comment by dmtrjsg** [comment]
Video workspace: 
- Removed the possibility of minting the NFT
- Action bar update (disabled buttons until there are changes to do)
- Cannot edit NFT - > Edit NFT option is removed


NFTs
- Mint your videos NFT and  Put NFT on Sale > https://www.figma.com/file/7LCeeLg9cCFSDAS4alvniX/NFT-Workspace?type=design&node-id=710%3A94795&mode=design&t=xLd9wHLPfDGXyngb-1
- Banners for mint NFT and put on sale

**Comment by dmtrjsg** [comment]
@KubaMikolajczyk morning sir, we've raised a small suggestion [here in Figma>](https://www.figma.com/file/7LCeeLg9cCFSDAS4alvniX/NFT-Workspace?type=design&node-id=1%3A15537&mode=design&t=ybwnpMchRmDO8rNX-1)

to address the case when assets are still not uploaded in full at the time when we show the first Modal.. pls have a look 🙏

**Comment by KubaMikolajczyk** [comment]
@dmtrjsg the comments you left in the figma were addressed and I explain my approach here in this quick video✅

🎥 Loom video: https://www.loom.com/share/3a79d2285da2475ca9ede64e568801c9?sid=e302fd96-638e-4cd6-b0d2-f49d1898b7ab

📄 My video changelog: https://www.figma.com/file/LF7LOMhOA46bRcYHWlw1Bq/My-videos?type=design&node-id=784-26965&mode=design&t=21Qne80D0QlfiNPw-4

📄 NFT workspace changelog: https://www.figma.com/file/7LCeeLg9cCFSDAS4alvniX/NFT-Workspace?type=design&node-id=1-14419&mode=design&t=jtJwIY6G2jsBBxyO-4

**Comment by dmtrjsg** [comment]
@KubaMikolajczyk good thinking, but there is a case when thumbnail gets uploaded with an error so re-upload is required. This can only be done via Edit video.  If that happens, it may not be obvious why NFT cannot be minted, but let's not overthink here and disabling the option to mint if thumbnail uploaded with error may be enough 👍 

cc @WRadoslaw

**Comment by KubaMikolajczyk** [comment]
Shouldnt we have an option saying reupload instead of edit video if that happens? ☝️ I wouldn't think about going to edit video if there is an error with my video

**Comment by traumschule** [comment]
> @traumschule NFTs are immutable on the runtime level, so not possible for the time being and comes with a long lead time..

Yes, i knew that and wrote `Please make it possible somehow.` - it needs a creative solution.

We cannot assume users understand the system and we have to expect a lot of frustration if the UI doesn't inform them well enough.

After all i heard here minting NFT right on creation almost always seems to be the wrong option.
Shouldn't it be the default to first create the asset, then add all secondary assets like thumbnail or transscript.
Only after everything is set and uploaded correctly, the user may be encouraged to have the option to turn it into a NFT.
Basically the `mint NFT` flag needs to have this big red warning that if anything is missing (not uploaded yet) there's a big risk in doing so and why.

**Comment by dmtrjsg** [comment]
@traumschule these cases are addressed in the designs here, please have a look and feel free to share your feedback: https://www.loom.com/share/3a79d2285da2475ca9ede64e568801c9?sid=e302fd96-638e-4cd6-b0d2-f49d1898b7ab

**Comment by traumschule** [comment]
Thanks!
- agree to disable actions until upload succeeded.
- Mint NFT dialogue only after upload makes sense.
- When user signaled to mint before upload, there could be a notification "Ready to mint your NFT now", to avoid premature freezing of unfinished uploads.
